### ISSUE TITLE
Is it possible to reference OTR_HTTPHOST by hostname instead of IP address?

### ISSUE THREAD

**Issue by begleysm** [owntracks/recorder#273]
The HTTPHOST line in my config looks like this: "OTR_HTTPHOST="192.168.1.50" and everything works fine.

The hostname "otrecorder" points to 192.168.1.50 (I can access the webpage at either "http://192.168.1.50:8083" or "http://otrecorder:8083").

Is it possible to set it up something like "OTR_HTTPHOST=otrecorder" so that IP assignments remain the task of the DHCP server and the recorder config stays more resilient to IP changes?  (I tried and failed so far).

**Comment by jpmens** [comment]
Yes, that certainly is possible. The value of `OTR_HTTPHOST` must be either an IP address (as you've been using) or a hostname which will be resolved by the server the Recorder is running on; I think you've shown that in your second paragraph.

The hostname to address mapping is performed by your DNS server which may or may not simultaneously be your DHCP server.

**Comment by begleysm** [comment]
I haven't been able to get this to work so far.

When I set it up as "OTR_HTTPHOST=192.168.1.50" I can access the web interface at either "http://192.168.1.50:8083" or "http://otrecorder:8083".

When I set it up as "OTR_HTTPHOST=otrecorder" I cannot access the web interface at either http://192.168.1.50:8083" or "http://otrecorder:8083".

**Comment by jpmens** [comment]
And what is the result of the command `ping otrecorder`, i.e. which IP address does that name map to both on your client and on the machine the Recorder is running on?

**Comment by begleysm** [comment]
You're on target.

When I ping "otrecorder" from another machine I get 192.168.1.50 as I would expect.  But when I ping "otrecorder" from the host I get 127.0.1.1 (it's on Debian).  So, as I'm sure you expected, setting "OTR_HTTPHOST=otrecorder" is basically setting it to point at my local loopback interface instead of eth0.

Not sure how to fix this but at least I understand the problem now.

**Comment by jpmens** [comment]
Either configure `192.168.1.50 otrecorder` in the `/etc/hosts` file on the host (warning: this could have side-effects for your particular setup) or create a new name set to that address in your DHCP/DNS server and use that.

**Comment by begleysm** [comment]
Thank you.  The /etc/hosts solution doesn't really work for me.  The whole point of this effort is to decouple the server configuration from the DNS/DHCP configuration as much as possible.  Moving a hardcoded IP address from the /etc/defaults/ot-recorder to /etc/hosts doesn't improve things (and as you say might cause other issues).

I'll look into adding a new nameset to the DHCP/DNS server.  In fact, right now, it is done by DD-WRT.  Moving to a standalone DNS server is on my (endless) to-do list.

As this isn't a problem with ot-recorder and I know the cause of the problem and a path to the solution.  This issue can probably be closed.  Thanks.
